odl/solvers/functional/functional.py

```diff
@@ -0,0 +1,1160 @@
+                                         cconj_f.grad_lipschitz))
+
+        self.__orig_cconj_f = cconj_f
+        self.__scaling = scaling
+
+    @property
+    def orig_cconj_f(self):
+        """The original convex conjugate functional."""
+        return self.__orig_cconj_f
+
+    @property
+    def scaling(self):
+        """The scaling."""
+        return self.__scaling
+
+    def _call(self, x):
+        """Apply the functional to the given point."""
+        return self.scaling * self.orig_cconj_f(x * (1.0 / self.scaling))
+
+    @property
+    def gradient(self):
+        """Gradient operator of the functional."""
+        return self.orig_cconj_f.gradient * (1.0 / self.scaling)
+
+    @property
+    def proximal(self):
+        """Return the proximal factory of the ConvexConjugateFuncScaling.
+
+        See Also
+        --------
+        proximal_arg_scaling
+        """
+        return lambda sigma: proximal_arg_scaling(self.orig_cconj_f.proximal,
+                                                  scaling=(1.0 / self.scaling)
+                                                  )(self.scaling * sigma)
+
+    @property
+    def convex_conj(self):
+        """Convex conjugate functional of the functional.
+
+        By the Fenchel-Moreau theorem this a scaling of the original
+        convex conjugate functional.
+        """
+        return self.scaling * self.orig_cconj_f.convex_conj
+
+
+class ConvexConjugateArgScaling(Functional):
+
+    """ The ``Functional`` representing ``(F( . * scaling))^*``.
+
+    This is a functional representing the conjugate functional of the
+    functional with scaled arguement: ``F(. * scaling)``. This is calculated
+    according to the rule
+
+        ``(F( . * scaling))^*(x) = F^*(x/scaling)``
+
+    where ``scaling`` is the scaling parameter. Note that this does not allow
+    for scaling with ``0``.
+
+    Parameters
+    ----------
+    convex_conj_f : `Functional`
+        Function corresponding to ``F^*``.
+
+    scaling : `float`, nonzero
+        The scaling parameter.
+
+    Notes
+    -----
+    The implementation assumes that the underlying  functional ``F`` is proper,
+    convex, and lower semi-continuous. Otherwise the convex conjugate of the
+    convex conjugate is not necessarily the functional itself.
+
+    For reference on the identity used, see [KP2015]_.
+    """
+
+    def __init__(self, cconj_f, scaling):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        convex_conj_f : `Functional`
+            Functional corresponding to ``F^*``.
+
+        scaling : 'float', nonzero
+            The scaling parameter.
+        """
+        if not isinstance(cconj_f, Functional):
+            raise TypeError('`cconj_f` {} is not a `Functional` instance'
+                            ''.format(cconj_f))
+
+        scaling = float(scaling)
+        if scaling == 0:
+            raise ValueError('Scaling with 0 is not allowed. Current value:'
```

Comment:
A question of intelligence vs. no intelligence. Obviously `_call` as it is now fails for `scaling=0`. So either we disallow that case or we handle it separately in `_call`.

The basic issue is that we need to decide if we want `FunctionalLeftScaling` with `scaling=0` to be `ZeroFunctional` or not. If yes, we will never get here with `scaling=0` unless we use the class explicitly.
The same holds for `ConstantFunctional`.